crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -279,66 +299,385 @@ where
+                    !matches!(
+                        x,
+                        Address::Internal(InternalAddress::IbcToken(_))
+                    )
+                })
+            })
+    }
+
+    // Apply the given send packet to the changed balances structure
+    fn apply_send_packet(
+        &self,
+        mut acc: ChangedBalances,
+        msg: &MsgTransfer,
+        keys_changed: &BTreeSet<Key>,
+    ) -> Result<ChangedBalances> {
+        // If a key change with the same port, channel, and commitment as this
+        // message cannot be found, then ignore this message
+        if self
+            .search_ibc_transfer(&msg.message, keys_changed)?
+            .is_none()
+        {
+            return Ok(acc);
+        };
+
+        // Since IBC denominations are derived from Addresses
+        // when sending, we have to do a reverse look-up of the
+        // relevant token Address
+        let Some(token) = Self::reverse_query_ibc_denom(
+            &msg.message.packet_data.token.denom,
+            &acc.ibc_denoms,
+        ) else {
+            return Ok(acc);
+        };
+        let delta = ValueSum::from_pair(
+            token.clone(),
+            Amount::from_uint(Uint(*msg.message.packet_data.token.amount), 0)
+                .unwrap(),
+        );
+        // If there is a transfer to the IBC account, then deduplicate the
+        // balance increase since we already accounted for it above
+        if is_sender_chain_source(
+            msg.message.port_id_on_a.clone(),
+            msg.message.chan_id_on_a.clone(),
+            &msg.message.packet_data.token.denom,
+        ) {
+            let post_entry =
+                acc.post.entry(addr_taddr(IBC)).or_insert(ValueSum::zero());
+            *post_entry =
+                checked!(post_entry - &delta).map_err(native_vp::Error::new)?;
+        }
+        // Required for the packet's receiver to get funds
+        let post_entry = acc
+            .post
+            .entry(ibc_taddr(msg.message.packet_data.receiver.to_string()))
+            .or_insert(ValueSum::zero());
+        // Enable funds to be received by the receiver in the
+        // IBC packet
+        *post_entry =
+            checked!(post_entry + &delta).map_err(native_vp::Error::new)?;
+
+        Ok(acc)
+    }
+
+    // Get the acknowledgement status associated with the given message
+    fn get_msg_ack(
+        &self,
+        message: &IbcMsgRecvPacket,
+        keys_changed: &BTreeSet<Key>,
+        ibc_acks: &BTreeMap<Vec<u8>, AcknowledgementStatus>,
+    ) -> Result<Option<AcknowledgementStatus>> {
+        // Ensure that the event corresponds to the current changes
+        // to storage
+        let ack_key = storage::ack_key(
+            &message.packet.port_id_on_a,
+            &message.packet.chan_id_on_a,
+            message.packet.seq_on_a,
+        );
+        if !keys_changed.contains(&ack_key) {
+            // Otherwise ignore this event
```

Comment:
What do you mean by "otherwise"? In this case there is no acknowledgement right?

Reply:
Yes, you're correct. Will rephrase this comment.